src/Functions/formatDateTime.cpp

```diff
@@ -272,6 +272,16 @@ class FunctionFormatDateTimeImpl : public IFunction
             writeNumber2(target + 6, ToSecondImpl::execute(source, timezone));
         }
 
+        static void timezoneOffset(char * target, Time source, const DateLUTImpl & timezone)
+        {
+            auto offset = TimezoneOffsetImpl::execute(source, timezone);
+            if (offset < 0)
+                *target = '-';
+
+            writeNumber2(target + 1, ToHourImpl::execute(offset, timezone));
```

Comment:
It is not necessary to use ToHourImpl, ToSecondImpl, because they have a different meaning.

It's enough to write `/ 3600`, `% 3600 / 60`.

Reply:
And to prove that it was wrong, let's add a test for some timezone that has offset of a fractional number of hours at 1970-01-01 00:00:00.

Reply:
Thanks for the feedback.